MdePkg/Library/BaseRngLib/AArch64/Rndr.c

```diff
@@ -43,15 +38,12 @@ BaseRngLibConstructor (
   VOID
   )
 {
-  UINT64  Isar0;
-
   //
   // Determine RNDR support by examining bits 63:60 of the ISAR0 register returned by
   // MSR. A non-zero value indicates that the processor supports the RNDR instruction.
   //
-  Isar0 = ArmReadIdIsar0 ();
-
-  mRndrSupported = ((Isar0 & RNDR_MASK) != 0);
+  mRndrSupported = !((ArmReadIdAA64Isar0Reg () >> ARM_ID_AA64ISAR0_EL1_AES_SHIFT) &

```

Comment:
This is using the wrong shift and mask, no?

Reply:
yes indeed ...